packages/backend/error/src/api-error.ts

```diff
@@ -1,99 +1,50 @@
 import { createDebug } from "@gravitywelluk/debug";
 
-const debug = createDebug("ERROR");
+import { ErrorType } from "./error-type";
 
-/** Available error codes */
-export enum ErrorType {
-  ApiError = "API_ERROR",
-  ApiConectionError = "API_CONNECTION_ERROR",
-  AuthenticationError = "AUTHENTICATION_ERROR",
-  ForbiddenError = "FORBIDDEN_ERROR",
-  InvalidData = "INVALID_DATA_ERROR",
-  DatabaseError = "DATABASE_ERROR",
-  NotFoundError = "NOT_FOUND_ERROR",
-  ThirdPartyError = "THIRD_PARTY_ERROR",
-  TooManyRequests = "TOO_MANY_REQUESTS",
-  UnknownError = "UNKNOWN_ERROR"
-}
+const debug = createDebug("ERROR");
 
-export interface ApiErrorResponse<C = unknown> {
-  /** Api response code */
-  statusCode: number;
-  /** Generic error type */
-  type: ErrorType;
+export interface ApiErrorResponse {
+  /** HTTP status code */
+  statusCode: ErrorType;
+  /** Title of HTTP status code */
+  title: string;
+  /** User-facing error message */
   message: string;
-  /** More specific error code, types specified from parent */
-  code?: C;
-  /** Extra more specific relevant information */
-  param?: Record<string, unknown>;
+  /** Additional relevant error details */
+  detail?: string;
 }
 
 /**
- * Standardised api error class
+ * API Error class, storing error type, message and additional details
  */
-export default class APIError<C> extends Error {
+export default class APIError extends Error {
   public readonly type: ErrorType;
-  public readonly code?: C;
-  public readonly param?: Record<string, unknown>;
+  public readonly detail?: string;
 
-  constructor(msg: string, type?: ErrorType, code?: C, param?: Record<string, unknown>) {
+  /**
+   * @param type The type of error to throw - each of these maps to a distinct HTTP status code
+   * @param msg The user-facing error message
+   * @param detail Any additional details about the error
+   */
+  constructor(type: ErrorType, msg: string, detail?: string) {
```

Comment:
Is there any benefit in the type being first? Extending of the error class, personally I think its nicer to keep the arguments in an order thats familiar to a default JavaScript Error. It would also make refactoring updating existing projects using the error module a much easier process. Find and replace would be possible rather which wouldnt be if the order of arguments switches.

Reply:
Sure - I'm happy either way. Only reason I flipped it was because I naturally think about picking the status code and _then_ the message - but happy to switch again for consistency with other JS errors!